src/xenia/base/testing/memory_test.cc

```diff
@@ -0,0 +1,392 @@
+/**
+ ******************************************************************************
+ * Xenia : Xbox 360 Emulator Research Project                                 *
+ ******************************************************************************
+ * Copyright 2015 Ben Vanik. All rights reserved.                             *
+ * Released under the BSD license - see LICENSE in the root for more details. *
+ ******************************************************************************
+ */
+
+#include "xenia/base/memory.h"
+
+#include "third_party/catch/include/catch.hpp"
+
+namespace xe {
+namespace base {
+namespace test {
+
+TEST_CASE("copy_128_aligned", "Copy and Swap") {
+  uint8_t src[256], dest[256];
+  for (uint8_t i = 0; i < 255; ++i) {
+    src[i] = 255 - i;
+  }
+  std::memset(dest, 0, sizeof(dest));
+  copy_128_aligned(dest, src, 1);
+  REQUIRE(std::memcmp(dest, src, 128));
+  REQUIRE(dest[128] == 0);
+
+  std::memset(dest, 0, sizeof(dest));
+  copy_128_aligned(dest, src, 2);
+  REQUIRE(std::memcmp(dest, src, 256));
+
+  std::memset(dest, 0, sizeof(dest));
+  copy_128_aligned(dest, src + 1, 1);
+  REQUIRE(std::memcmp(dest, src + 1, 128));
+}
+
+TEST_CASE("copy_and_swap_16_aligned", "Copy and Swap") {
+  uint16_t a[8] = {0x1111};
```

Comment:
As for the test cases with src/dest on the stack, I think the `alignas(16)` keyword would work.